src/handlers.rs

```diff
@@ -279,4 +394,102 @@ mod test {
     }
+
+    #[test]
+    fn test_handle_request_getai() {
+        let request = protocol::Request {
+            ty: protocol::RequestType::GETAI,
+            key: &CString::new("localhost".to_string())
+                .unwrap()
+                .into_bytes_with_nul(),
+        };
+
+        // The getaddrinfo addresses ordering is quite unpredictable.
+        // Trying the two potential orderings.
+        // We could also be on a IPv4-only machine on CI.
+        let gen_ai_resp = |addrs| protocol::AiResponse {
+            addrs,
+            canon_name: "localhost".to_string(),
+        };
+        let ai_resp_1 = gen_ai_resp(vec![
+            IpAddr::new_v6(0, 0, 0, 0, 0, 0, 0, 1),
+            IpAddr::new_v4(127, 0, 0, 1),
+        ]);
+        let ai_resp_2 = gen_ai_resp(vec![
+            IpAddr::new_v4(127, 0, 0, 1),
+            IpAddr::new_v6(0, 0, 0, 0, 0, 0, 0, 1),
+        ]);
+        let ai_resp_3 = gen_ai_resp(vec![IpAddr::new_v4(127, 0, 0, 1)]);
+        let expected_1: Vec<u8> = serialize_address_info(&ai_resp_1)
+            .expect("serialize_address_info should serialize the addresseses");
+        let expected_2: Vec<u8> = serialize_address_info(&ai_resp_2)
+            .expect("serialize_address_info should serialize the addresseses");
+        let expected_3: Vec<u8> = serialize_address_info(&ai_resp_3)
+            .expect("serialize_address_info should serialize the addresseses");
+
+        let output =
+            handle_request(&test_logger(), &request).expect("should handle request with no error");
+
+        assert!(
+            expected_1 == output || expected_2 == output || expected_3 == output,
+            "\nExpecting \n{:?}\nTo be equal to\n{:?}\nor\n{:?}\nor\n{:?}\n",
+            output,
+            expected_1,
+            expected_2,
+            expected_3
+        );
+    }
+
+    #[test]
+    fn test_handle_request_getai_no_results() {
+        let request = protocol::Request {
+            ty: protocol::RequestType::GETAI,
+            key: &CString::new("dontexist".to_string())
+                .unwrap()
+                .into_bytes_with_nul(),
+        };
+
+        let expected = protocol::AI_RESPONSE_HEADER_NOT_FOUND.as_slice();
+        let output =
+            handle_request(&test_logger(), &request).expect("should handle request with no error");
+
+        assert_eq!(expected, output);
+    }
+
+    /// This test compares the [serialize_adress_info] output to a
+    /// byte array generated by the original nscd implementation.
+    #[test]
+    fn test_serialize_address_info() {
+        let test_response = AiResponse {
+            addrs: vec![
+                IpAddr::new_v6(0, 0, 0, 0, 0, 0, 0, 1),
+                IpAddr::new_v4(127, 0, 0, 1),
+            ],
+            canon_name: "localhost".to_string(),
+        };
+
+        let expected: Vec<u8> = vec![
+            // Response Header
+            // ===============
+            // version number = 2
```

Comment:
The rustfmt did fold the comments up into the wrong line.

Reply:
Aha, thanks!